### ISSUE TITLE
Asserts relating to range in ClassUnloadAssumption.cpp in debug build

### ISSUE THREAD

**Issue by caohaley** [eclipse/openj9#5843]
This assert comes up eventually after you comment out the other asserts before it (the asserts in issue #5841 and the asserts mentions in issue #5631 ):

```
Assertion failed at ../compiler/runtime/ClassUnloadAssumption.cpp:1544: prev.getEnd() < cur.getStart()
	UAR: ERROR: range 56 [0x321dead - 0x321dead] overlaps previous range ending at 0x321dead
```
May be indicative of existing bugs.

**Comment by caohaley** [comment]
@ymanton @mpirvu  could you add the JitaaS tag to this?

**Comment by mpirvu** [comment]
Fixed by #6285
